src/aead/aead.rs

```diff
@@ -25,6 +25,82 @@
 //! [AEAD]: http://www-cse.ucsd.edu/~mihir/papers/oem.html
 //! [`crypto.cipher.AEAD`]: https://golang.org/pkg/crypto/cipher/#AEAD
 
+//!
+//! **Example (if you don't know what you are doing, read this first)**
+//!
+//!```rust
+//!
+//!extern crate ring;
+//!
+//!use ring::aead::*;
+//!use ring::pbkdf2::*;
+//!use ring::rand::SystemRandom;
+//!
+//!fn main() {
+//!    // The password will be used to generate a key and it is normally used when
+//!    // it is generated by the user. If you need a key which will be stored in a file you
+//!    // generate it directly.
+//!    let password = b"nice password";
+//!
+//!    // Usually the salt has some random data and something that relates to the user
+//!    // like an username and it must be 8 bytes (64 bits)
+//!    let salt = [0, 1, 2, 3, 4, 5, 6, 7];
+//!
+//!    // Keys must have 32 bytes and if you want to use an user password for it you have
+//!    // to use an algorithm to derive it (in this case pbkdf2).
+//!    let mut key = [0; 32];
+//!    derive(&HMAC_SHA256, 100, &salt, &password[..], &mut key);
+//!
+//!    // Your private data
+//!    let content = b"content to encrypt".to_vec();
+//!    println!("Content to encrypt's size {}", content.len());
+//!
+//!    // Additional data that you would like to send and it would not be encrypted but it
+//!    // would be signed
+//!    let additional_data: [u8; 0] = [];
+//!
+//!    // Ring uses the same input variable as output
+//!    let mut in_out = content.clone();
+//!
+//!    // The input/output variable need some space to store the tag which will have the
+//!    // signature
+//!    println!("Tag len {}", CHACHA20_POLY1305.tag_len());
+//!    for _ in 0..CHACHA20_POLY1305.tag_len() {
+//!        in_out.push(0);
+//!    }
+//!
+//!    // Opening key used to decrypt data
+//!    let opening_key = OpeningKey::new(&CHACHA20_POLY1305, &key).unwrap();
+//!
+//!    // Sealing key used to encrypt data
+//!    let sealing_key = SealingKey::new(&CHACHA20_POLY1305, &key).unwrap();
+//!
+//!    // Random data must be used only once per encryption and ideally if you encrypt blocks
+//!    // it should be generated on per block see an example of poorly generated nonces
+//!    // where a new nonce is generated at every few pixels at
+//!    // https://github.com/jaysonsantos/bad-encryption-example
+//!    // it must be 12 bytes (96 bits) and you should use a secure random generator, not
+//!    // some pseudo random algorithm without entropy like /dev/random.
+//!    let mut nonce = vec![0; 12];
+//!
+//!    // Fill nonce with random data
+//!    let rand = SystemRandom::new();
+//!    rand.fill(&mut nonce).unwrap();
+//!
+//!    // Encrypt data into in_out variable
+//!    let output_size = seal_in_place(&sealing_key, &nonce, &additional_data, &mut in_out,
+//!                                    CHACHA20_POLY1305.tag_len()).unwrap();
+//!
```

Comment:
The tag's length is not guaranteed to be `out_suffix_capacity`. So it might be that you `resize()` the vector from 80 to 100 to make room for the tag, but then it turns out the tag is only 15 long, so your ciphertext is 95. That's the number returned from `seal_in_place`.

I don't know if that difference actually ever occurs -- it might well be that `output_size` always matches the size of the vector, so a `.truncate()` call wouldn't do anything at all. Nevertheless, that's how the API is designed (if I understand it correctly, that is).

So now, AFAIK you wouldn't loose the tag, it's included in that size. Just try out, if the decryption fails, something went wrong, otherwise it's correct.